fabtests/functional/rdm_tagged_peek.c

```diff
@@ -330,7 +309,19 @@ static int run(void)
 			return ret;
 		}
 	}
-
+	/*
+	 * We are not following the rx_seq in common code, so some extra
+	 * procedure is needed here.
+	 * ft_finalize() will post a send and its tag (the tx_seq here)
+	 * should match the tag used by the last recv posted by ft_rx, which
+	 * should be rx_seq - 1 (the rx_cq_cntr at that time).
+	 * And the rx_cq_cntr should also be set to rx_seq - 1 so ft_get_rx_comp()
+	 * will poll 1 cqe.
+	 */
+	tx_seq = rx_seq - 1 ;
+	tx_cq_cntr = tx_seq;
+	rx_cq_cntr = rx_seq - 1;
+	ft_tag = 0;
```

Comment:
Can we somehow not need the above code?  This is a difficulty for maintainability.  If there's no easy way around it for now, we can come back someday to try to address.

Reply:
Removed this hack in latest revision.